File: examples/nodejs/01_basic_parse.mjs
// Basic: Parse, tokenize, and validate QQL queries.
import { parse, tokenize, isValid } from 'nqql';

// Parse a CREATE COLLECTION statement
const ast = parse('CREATE COLLECTION docs HYBRID WITH HNSW (m = 32)');
console.log('=== Parsed AST ===');
console.log(ast.substring(0, 500));
console.log();

// Tokenize a QUERY
const tokens = JSON.parse(tokenize("QUERY 'vector database' FROM docs LIMIT 10"));
console.log('=== Tokens ===');
for (const t of tokens) {
  console.log(`  ${t.kind.padEnd(12)} ${(t.text).padEnd(30)}  pos=${t.pos}`);
}
console.log();

// Validate queries
for (const q of [
  "QUERY 'hello' FROM docs LIMIT 5",
  'CREATE COLLECTION docs',
  'SELECT * FROM docs WHERE id = 1',
  '',
  'BOGUS STUFF',
]) {
  const valid = isValid(q);
  console.log(`  valid=${valid}  ${q}`);
}

File: examples/python/01_basic_parse.py
"""Basic: Parse, tokenize, and validate QQL queries."""
import pyqql

# Parse a CREATE COLLECTION statement
ast = pyqql.parse("CREATE COLLECTION docs HYBRID WITH HNSW (m = 32)")
print("=== Parsed AST ===")
print(ast[:500])
print()

# Tokenize a QUERY
tokens = pyqql.tokenize("QUERY 'vector database' FROM docs LIMIT 10")
print("=== Tokens ===")
for t in tokens:
    print(f"  {t['kind']:12} {t['text']!r:30}  pos={t['pos']}")
print()

# Validate queries
for q in [
    "QUERY 'hello' FROM docs LIMIT 5",
    "CREATE COLLECTION docs",
    "SELECT * FROM docs WHERE id = 1",
    "",
    "BOGUS STUFF",
]:
    valid = pyqql.is_valid(q)
    print(f"  valid={valid}  {q!r}")
